npm/webpack-dev-server/src/helpers/angularHandler.ts

```diff
@@ -0,0 +1,237 @@
+  return {
+    outputPath: 'dist/angular-app',
+    assets: [],
+    styles: [],
+    scripts: [],
+    budgets: [],
+    fileReplacements: [],
+    outputHashing: 'all',
+    inlineStyleLanguage: 'css',
+    stylePreprocessorOptions: { includePaths: [] },
+    resourcesOutputPath: undefined,
+    commonChunk: true,
+    baseHref: undefined,
+    deployUrl: undefined,
+    verbose: false,
+    progress: false,
+    i18nMissingTranslation: 'warning',
+    i18nDuplicateTranslation: 'warning',
+    localize: undefined,
+    watch: true,
+    poll: undefined,
+    deleteOutputPath: true,
+    preserveSymlinks: undefined,
+    showCircularDependencies: false,
+    subresourceIntegrity: false,
+    serviceWorker: false,
+    ngswConfigPath: undefined,
+    statsJson: false,
+    webWorkerTsConfig: undefined,
+    crossOrigin: 'none',
+    allowedCommonJsDependencies: [], // Add Cypress 'browser.js' entry point to ignore "CommonJS or AMD dependencies can cause optimization bailouts." warning
+    buildOptimizer: false,
+    optimization: false,
+    vendorChunk: true,
+    extractLicenses: false,
+    sourceMap: true,
+    namedChunks: true,
+    ...projectConfig.architect.build.options,
+    ...projectConfig.architect.build.configurations?.development || {},
+    tsConfig,
+    aot: false,
+  }
+}
+
+export async function generateTsConfig (devServerConfig: WebpackDevServerConfig, projectConfig: AngularJsonProjectConfig): Promise<string> {
+  const { cypressConfig } = devServerConfig
+  const { projectRoot } = cypressConfig
+
+  const specPattern = Array.isArray(cypressConfig.specPattern) ? cypressConfig.specPattern : [cypressConfig.specPattern]
+
+  const getProjectFilePath = (...fileParts: string[]): string => toPosix(path.join(projectRoot, ...fileParts))
+
+  const includePaths = [...specPattern.map((pattern) => getProjectFilePath(pattern))]
+
+  if (cypressConfig.supportFile) {
+    includePaths.push(toPosix(cypressConfig.supportFile))
+  }
+
+  if (projectConfig.architect.build.options.polyfills) {
+    const polyfills = getProjectFilePath(projectConfig.architect.build.options.polyfills)
+
+    includePaths.push(polyfills)
+  }
+
+  const cypressTypes = getProjectFilePath('node_modules', 'cypress', 'types', 'index.d.ts')
+
+  includePaths.push(cypressTypes)
+
+  const tsConfigContent = JSON.stringify({
+    extends: getProjectFilePath('tsconfig.json'),
+    compilerOptions: {
+      outDir: getProjectFilePath('out-tsc/cy'),
+      allowSyntheticDefaultImports: true,
+    },
+    include: includePaths,
+  })
+
+  const tsConfigPath = path.join(await getTempDir(), 'tsconfig.json')
+
+  await fs.writeFile(tsConfigPath, tsConfigContent)
+
+  return tsConfigPath
+}
+
+export async function getTempDir (): Promise<string> {
+  const cypressTempDir = path.join(tmpdir(), 'cypress-angular-ct')
+
+  await fs.ensureDir(cypressTempDir)
+
+  return cypressTempDir
+}
+
+export async function getAngularCliModules (projectRoot: string) {
```

Comment:
Angular 13 and 14 share the same API, but if we wanted to support Angular 12 we would need to change this. Need to look into current Angular 12 usages.

Reply:
12.2 (most popular 12.x version)

![image](https://user-images.githubusercontent.com/19196536/180389056-080f4abd-1c7b-4849-84d8-a8f9a1277125.png)

Seems on par with 13.x and 14.x Definitely good to ship with some and look to add more if needed - we can set in launchpad what version(s) we support, for now >= ^13 <= ^14